src/app/report/edit-display/edit-display.component.ts

```diff
@@ -107,30 +116,37 @@ export class EditDisplayComponent {
   }
 
   downloadReport(exportBinary: boolean, exportXML: boolean): void {
-    let queryString: string = this.report.xml ? this.report.storageId.toString() : this.report.uid.split('#')[0];
-    this.helperService.download(queryString + '&', this.currentView.storageName, exportBinary, exportXML);
+    const queryString: string = this.report.xml ? this.report.storageId.toString() : this.report.uid.split('#')[0];
```

Comment:
It feels wrong to check if the XML attribute exists to then use the storageId. What is it asserting? Would it otherwise be a different type of repost? Having a type for this would be really nice.

Reply:
I did not change this, just made it a const

Reply:
I know, but i reviewed it

Reply:
Well to answer your question, this.report can be the root level report or one of its checkpoints, thats why that assertion is done.

Reply:
I'm not getting that from the code

Reply:
Well than make an issue, I'm changing it in this PR